core/src/TopNavDropDown.html

```diff
@@ -4,7 +4,11 @@
   <ul class="fd-menu__list fd-menu__list--top">
     {#if children}
     {#each children as node}
-    <li on:click="onActionClick(node)" data-e2e="topnav-dropdown-item">
+    <li
+      on:click="onActionClick(node)"
+      data-e2e="topnav-dropdown-item"
+      data-testid="{ node.testId ? node.testId : prepareForTests(node.pathSegment, node.label)}"
```

Comment:
Aince it is always the same if else construct, how about a small helper `getTestId(node)`?
Could be done with refactoring `prepareForTests` by passing node and doing the complete logic there.